### PR TITLE
basic framework for casbin-raft, use etcd raft

### PR DESCRIPTION
**Pull request by dovics**

### CODE REVIEW

File: integration/ca.crt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+-----BEGIN CERTIFICATE-----
```
Comment (hsluoyz) [inline integration/ca.crt]: Why upload SSH keys here?

**Review by hsluoyz — COMMENTED** [review]
Why upload SSH keys here?

File: integration/ca.crt (hunk 1 above)
Comment (dovics) [reply integration/ca.crt]: This is only used to test the transport with TLS, I can remove this if you think it is unnecessary

File: node.go [hunk 2]
```diff
@@ -142,37 +162,43 @@ func (n *Node) Restart() error {
 	n.snapshotter = snap.New(n.snapdir)
 	snapshot, err := n.snapshotter.Load()
 	if err != nil && err != snap.ErrNoSnapshot {
-		return fmt.Errorf("casbin: error loading snapshot (%v)", err)
+		return fmt.Errorf("casbin: Failed loading snapshot (%v)", err)
```
Comment (nodece) [inline node.go]: I recommend use `https://github.com/pkg/errors`.

The code like `errors.Wrap(err, "Failed loading snapshot")`.

Comment (dovics) [reply node.go]: > I recommend use `https://github.com/pkg/errors`.
> 
> The code like `errors.Wrap(err, "Failed loading snapshot")`.

it likes good, I will try

**Pull request merged**